src/borg/repository.py

```diff
@@ -482,7 +482,7 @@ def complete_xfer(intermediate=True):
                     shadowed_put_exists = key not in self.shadow_index or any(
```

Comment:
Since we only track PUTs and we have never "just delete" a segment containing puts.